src/dom/DomEvent.DoubleTap.js

```diff
@@ -33,7 +33,7 @@ L.extend(L.DomEvent, {
 		}
 
 		function onTouchEnd() {
-			if (doubleTap) {
+			if (doubleTap && !touch.cancelBubble) {

```

Comment:
Apparently not, because onTouchStart sets touch to an event. There might be a degenerate case where the user starts the touch outside the map, then finishes it inside the map. Anyway the current code (a few lines down) assumes that `touch` exists and is an object, so no regression here.

Maybe the right thing to do is to follow the pattern of the `preclick` event in the `map-domevent` branch - fire a synthetic event which resembles a DOM event, instead of a leaflet event (which includes coordinates).

Right now I'd rather patch the bug and not think of any more refactoring.